source/memoizeWith.js

```diff
@@ -31,6 +32,27 @@ import _has from './internal/_has';
  *      factorial(5); //=> 120
  *      factorial(5); //=> 120
  *      count; //=> 1
+ *
+ * @example
+ *
+ *      let rid = 0;
+ *      const refMap = new WeakMap();
+ *      const refLookup = R.ifElse(
+ *          refMap.has,
+ *          refMap.get,
+ *          arg => (refMap.set(arg, ++rid), rid)
+ *      );
+ *      const weakReferenceId = R.compose(R.join(','), R.map(refLookup))
+ *
+ *      const memoizeByRef = R.memoizeWith(function() {
+ *          return weakReferenceId(arguments);
+ *      });
+ *
+ *      // getPrimaryContact :: [{primary: bool}] -> {primary: true}
+ *      const getPrimaryContact = memoizeByRef(R.compose(
+ *          R.find(R.propEq('primary', true)),
+ *          R.prop('contacts')
+ *      ));
```

Comment:
What about a simpler example? The one you have is confusing to me, and looks like it's replacing the memoizing capabilities of the function.

If you have a large object, it would be expensive to call `toString` on it. You can use this function to provide a separate version of equality. Here is a very contrived example that should probably be cleaned up a bit:

```
// very contrived
const objFib = (obj) => 
  obj.val <= 1 ? 
  1 : 
  objFib(evolve({val: subtract(__, 1)}, obj)) + objFib(evolve({val: subtract(__, 2)}, obj));
//now we only have to deal with the one property that matters instead of the whole object
const memoVal = compose(toString, prop('val'));
const memoized = memoizeWith(memoVal, objFib);
```

Reply:
I'm open to ideas.  The intent is to demonstrate the responsibilities/concerns of the key generating function without getting bogged down in them.  Your suggestion leads me to think about memoizing a function operating on database records by indexing on the record id.  But like the pre-existing example, it obscures variadic arguments, reference vs value, mutating input, etc.

My hope is that the user will be informed enough to use `memoizeWith` responsibly, alerting them to the pitfalls while simultaneously give the user a reason to take on those concerns rather than blindly use `memoize` or `memoizeWith(toString)`.

Maybe I'm just trying to do too much?

Reply:
I did just notice my signature didn't match my implementation, so I reduced the implementation to the signature.  I don't think it really addresses your concerns though.

Perhaps I should reduce to just the description changes and ditch the new examples.